[stable/redis-ha] Update Redis to 5.0.5

arbourd opened the pull request:
## What this PR does / why we need it:

Updates Redis to 5.0.5 which fixes a few critical bugs:
https://raw.githubusercontent.com/antirez/redis/5.0/00-RELEASENOTES

#### Special notes for your reviewer:

~I set the Chart version to be `3.7.0` assuming PR #15030 will go in first. Please make sure it is, or amend this version, before merging.~

#### Checklist

- [x] [DCO](https://github.com/helm/charts/blob/master/CONTRIBUTING.md#sign-your-work) signed
- [x] Chart Version bumped
- [x] Variables are documented in the README.md
- [x] Title of the PR starts with chart name (e.g. `[stable/chart]`)

ssalaues commented on stable/redis-ha/Chart.yaml:
```diff
@@ -6,8 +6,8 @@ keywords:
 - database
-version: 3.6.0
-appVersion: 5.0.3
+version: 3.7.0
```
@arbourd I think because it's a patch release of Redis, you could just bump the patch version of the chart as well.

arbourd replied:
Sounds good! Done :)

Commit:
[stable/redis-ha] Update Redis to 5.0.5

Release notes:
https://raw.githubusercontent.com/antirez/redis/5.0/00-RELEASENOTES

Signed-off-by: Dylan Arbour <dylan.arbour@gmail.com>
